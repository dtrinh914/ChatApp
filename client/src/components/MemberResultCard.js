import React from 'react';
import MemberItem from './MemberItem';
import ListItem from '@material-ui/core/ListItem';
import {makeStyles} from '@material-ui/styles';
import Button from '@material-ui/core/Button'
import IconButton from '@material-ui/core/IconButton';
import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline';
import DoneIcon from '@material-ui/icons/Done';


const useStyles = makeStyles({
    root:{
        display:'flex',
        alignItems: 'center',
        justifyContent: 'space-between'
    },
    add:{
        fontSize: '1rem'
    }
});

export default function MemberResultCard({username, userId, status, sendInvite}) {
    const classes = useStyles();

    const handleClick = () => {
        sendInvite(userId);
    }
    let statusButton;

    switch(status){
        case 'active':
            statusButton = <DoneIcon data-testid='active-icon' />
            break;
        case 'pending':
            statusButton = <Button data-testid='pending-icon'>Pending</Button>
            break;
        default:
            statusButton = <IconButton size='small' onClick={handleClick} data-testid='add-button'>
                                <AddCircleOutlineIcon className={classes.add} />
                            </IconButton>
            break;
    }
                        

    return (
        <ListItem data-testid='member-result-card' className={classes.root}>
            <MemberItem username={username} />
            {statusButton}
        </ListItem>
    )
}
