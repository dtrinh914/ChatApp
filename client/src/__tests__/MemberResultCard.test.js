import React from 'react';
import MemberResultCard from '../components/MemberResultCard';
import {cleanup, getByTestId, fireEvent, render} from '@testing-library/react';

afterEach(cleanup);

describe('<MemberResultCard /> with an active member', ()=>{
    it('should match snapshot', ()=>{
        const {container} = render(<MemberResultCard username='test_user' status='active' />);
        expect(container).toMatchSnapshot();
    })
});

describe('<MemberResultCard /> with a pending member', ()=>{
    it('should match snapshot', ()=>{
        const {container} = render(<MemberResultCard username='test_user' status='pending' />)
        expect(container).toMatchSnapshot();
    })
});

describe('<MemberResultCard /> with any other person', ()=>{
    const mockSendInvite = () => {console.log('Triggered')};

    it('should match snapshot', ()=>{
        const {container} = render(<MemberResultCard username='test_user' 
                                    userId='test_id' status='add' sendInvite={mockSendInvite} />)
        expect(container).toMatchSnapshot();
    })

    it('should trigger the sendInvite function',()=>{
        const {container} = render(<MemberResultCard username='test_user' 
                                    userId='test_id' status='add' sendInvite={mockSendInvite} />)
        const addButton = getByTestId(container, 'add-button');
        const spy = jest.spyOn(console, 'log');
        
        fireEvent.click(addButton);
        expect(spy).toHaveBeenCalled();
    });
});