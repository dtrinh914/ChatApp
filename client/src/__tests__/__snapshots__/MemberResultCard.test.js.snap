// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemberResultCard /> with a pending member should match snapshot 1`] = `
<div>
  <li
    class="MuiListItem-root makeStyles-root-32 MuiListItem-gutters"
    data-testid="member-result-card"
  >
    <div
      class="makeStyles-user-46"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-45 MuiAvatar-colorDefault"
      >
        T
      </div>
      <span>
        test_user
      </span>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      data-testid="pending-icon"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Pending
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </li>
</div>
`;

exports[`<MemberResultCard /> with an active member should match snapshot 1`] = `
<div>
  <li
    class="MuiListItem-root makeStyles-root-1 MuiListItem-gutters"
    data-testid="member-result-card"
  >
    <div
      class="makeStyles-user-15"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-14 MuiAvatar-colorDefault"
      >
        T
      </div>
      <span>
        test_user
      </span>
    </div>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      data-testid="active-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
      />
    </svg>
  </li>
</div>
`;

exports[`<MemberResultCard /> with any other person should match snapshot 1`] = `
<div>
  <li
    class="MuiListItem-root makeStyles-root-96 MuiListItem-gutters"
    data-testid="member-result-card"
  >
    <div
      class="makeStyles-user-110"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-109 MuiAvatar-colorDefault"
      >
        T
      </div>
      <span>
        test_user
      </span>
    </div>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
      data-testid="add-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-add-97"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </li>
</div>
`;
